drivers/gpu/drm/i915/i915_active.c

```diff
@@ -130,6 +131,57 @@ static inline void debug_active_assert(struct i915_active *ref) { }
 
 #endif
 
+#if defined(__FreeBSD__)
+static struct llist_head linux_kmem_cache_list = LLIST_HEAD_INIT();
+
+struct cache_addr {
+	struct linux_kmem_cache *cache;
+	struct llist_node *member;
+};
+
+static void
+linux_kmem_cache_free_async_fn(struct irq_work *irqw)
+{
+	struct llist_node *freed_addr;
+
+	while ((freed_addr = llist_del_first(&linux_kmem_cache_list)) != NULL) {
+		struct cache_addr *c = container_of(&freed_addr, struct cache_addr, member);
+		kmem_cache_free(c->cache, freed_addr);
```

Comment:
freed_addr points to member of freed `struct cache_addr`, rather than to structure itself. Change this line to
```
kmem_cache_free(c->cache, c);
```